Clarified ISSUE_TEMPLATE.md with windows arch instructions

coderjoe opened the pull request:
This commit updates the ISSUE_TEMPLATE.md file with more debugging
instructions for users in windows and prompts users for more information
when submitting issues.

Fixes #389

coderjoe commented:
@metaskills this seems overly simplistic, but I feel like this covers the majority of the things you might want or need people to do or collect before submitting an issue.

metaskills commented on ISSUE_TEMPLATE.md:
```diff
@@ -5,8 +5,10 @@ Having problems? Have you checked these first:
 * If not, please update then uninstall the TinyTDS gem and re-install it.
 * Using Ubuntu? If so, you may have forgotten to install FreeTDS first.
 * Doing work with threads and the raw client? Use the ConnectionPool gem?
+* On Windows? If so, do you need devkit for your ruby install?
```
Can we move this to the first line? Idea being put these in order we think are most issues first.

Can we change the first line from `Having problems? Have you checked these first:` to something like this with BOLD markdown?

**PLEASE CHECK THESE FIRST**

metaskills commented on ISSUE_TEMPLATE.md:
```diff
@@ -5,8 +5,10 @@ Having problems? Have you checked these first:
 * What platform you are on. Windows, Mac, Ubuntu, etc.
-* What version of FreeTDS you are using.
+* What version of TinyTDS are you using?
+* What version of FreeTDS you are using?
```
Maybe we have commands to run here after the question. Like:

Run `tsql -C` to see. Optionally show output.

coderjoe replied:
These are all good ideas. I'll make these changes and update either tonight or tomorrow. 👍

coderjoe commented:
I've re-ordered the "check this first" section according to a completely non-scientific search of issue frequency by eyeballing the last 100 issues or so. I've made it bold and renamed the header to be more commanding to "before submitting an issue check these first!". I've also added explicit commands, where possible, to help users collect useful information about their architecture and software versions.

metaskills commented:
Love this new format :)
